src/kit/LineChart.tsx

```diff
@@ -87,18 +90,15 @@ export const LineChart: React.FC<LineChartProps> = ({
     [series],
   );
 
-  const seriesNames: string[] = useMemo(() => {
-    return series.map((s) => {
-      return metricToStr({ group: s.metricType ?? 'unknown', name: s.name ?? 'unknown' });
-    });
-  }, [series]);
-
   const chartData: AlignedData = useMemo(() => {
     const xSet = new Set<number>();
     const yValues: Record<string, Record<string, number | null>> = {};
 
     series.forEach((serie, serieIndex) => {
       yValues[serieIndex] = {};
+      if (hiddenSeries[serieIndex]) {
+        return;
```

Comment:
if we return without value here, it doesn't match the `AlignedData` type, does it?

Reply:
this part only returns the `forEach` and prevents adding anything to `xSet` or `yValues[serieIndex]`

Reply:
ok i got the intention of the code

[Mozilla doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) says 

> The forEach() method is [generic](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#generic_array_methods). It only expects the this value to have a length property and integer-keyed properties.
>
> There is no way to stop or break a forEach() loop other than by throwing an exception. If you need such behavior, the forEach() method is the wrong tool.

so I recommend you to use `for in` or `for` instead of `forEach`, and we can replace `return` with `continue`

Reply:
I would read "stop or break [..] the loop" to mean if I wanted to avoid all future series (i.e. `yValues[serieIndex + 1]`) but I only need to skip/continue this one series when it's hidden by a legend toggle.

Reply:
i know, but i just think the code would be more flexible for the future change. i'm fine with the current one